File: app/api/chat/utils/sanitize.ts
// File: app/api/chat/utils/sanitize.ts

/**
 * Sanitizes input text by escaping special characters.
 * @param input - The input text to be sanitized.
 * @returns {string} - The sanitized string.
 */
export function sanitizeInput(input: string): string {
  return input
    .replace(/&/g, "&amp;")
    .replace(/</g, "&lt;")
    .replace(/>/g, "&gt;")
    .replace(/"/g, "&quot;")
    .replace(/'/g, "&#039;");
}

File: app/api/chat/utils/sanitize.py
# File: app/api/chat/utils/sanitize.py

import re
import logging

# Set up a logger
logger = logging.getLogger(__name__)

def sanitize_input(input: str) -> str:
    sanitized = re.sub(r"[`~!@#$%^&*()_|+\-=?;:'\",.<>\{\}\[\]\\\/]", '', input)
    # Uncomment the line below to enable logging
    # logger.debug(f"app/api/chat/utils/sanitize.py - Sanitized input: {sanitized}")
    return sanitized
